File: app/opensearch/semantic_search.py
from opensearch.query_builder import QueryRequest, normalize_request


def build_semantic_query(req: QueryRequest) -> dict:
    req = normalize_request(req)
    return {
        "size": req.size,
        "query": {
            "knn": {
                "item_vector": {
                    "vector": req.vector,
                    "k": req.size,
                }
            }
        },
    }

File: backend/opensearch/semantic_search.go
package opensearch

func BuildSemanticQuery(req QueryRequest) map[string]any {
	req = NormalizeRequest(req)
	return map[string]any{
		"size": req.Size,
		"query": map[string]any{
			"knn": map[string]any{
				"item_vector": map[string]any{
					"vector": req.Vector,
					"k":      req.Size,
				},
			},
		},
	}
}
